aws/table_aws_eks_node_group.go

```diff
@@ -0,0 +1,257 @@
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
+func tableAwsEksNodeGroup(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_eks_node_group",
+		Description: "AWS Elastic Kubernetes Service Node Group",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"name", "cluster_name"}),
+			Hydrate:    getEksNodeGroup,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listEksClusters,
+			Hydrate:       listEksNodeGroup,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name associated with an Amazon EKS managed node group.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("NodegroupName"),
+			},
+			{
+				Name:        "node_group_arn",
+				Description: "The Amazon Resource Name (ARN) of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+				Transform:   transform.FromField("nodegroupArn"),
+			},
+			{
+				Name:        "cluster_name",
+				Description: "The Time when the cluster was created.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "version",
+				Description: "The Kubernetes version of the managed node group.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "release_version",
+				Description: "If the node group was deployed using a launch template with a custom AMI, then this is the AMI ID that was specified in the launch template. For node groups that weren't deployed using a launch template, this is the version of the Amazon EKS optimized AMI that the node group was deployed with.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "created_at",
+				Description: "The timestamp when the managed node group was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "modified_at",
+				Description: "The timestamp when the managed node group was last modified.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "status",
+				Description: "The current status of the managed node group.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "capacity_type",
+				Description: "The current status of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEksNodeGroup,
+			},
+			{
+				Name:        "scaling_config",
+				Description: "The scaling configuration details for the Auto Scaling group that is associated with node group.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEksNodeGroup,
```

Comment:
We should try to keep all the JSON fields at end of the table, please change accordingly